### DISCUSSION TITLE
Better `podman run` error message?

### DISCUSSION THREAD

**Discussion by osalbahr** [containers/podman#18426] [Ideas]
```
~$ podman run -it alpine
Error: failed to connect: dial tcp [::1]:49704: connect: connection refused
~$ podman machine start
Starting machine "podman-machine-default"
Waiting for VM ...
Mounting volume... /Users:/Users
Mounting volume... /private:/private
Mounting volume... /var/folders:/var/folders
API forwarding listening on: /var/run/docker.sock
Docker API clients default to this address. You do not need to set DOCKER_HOST.

Machine "podman-machine-default" started successfully
~$ podman run -it alpine
/ # uname -a
Linux 404bce48530f 6.2.9-300.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Mar 30 22:32:58 UTC 2023 x86_64 Linux
/ # exit
~$ 
```

Would it be a good idea to suggest running `podman machine start` when `podman run` fails? Maybe similar to [init.go#L178](https://github.com/containers/podman/blob/f11ba8dc8ca2c7fddfc610978477af9438736220/cmd/podman/machine/init.go#L178). I find `Error: failed to connect: dial tcp [::1]:49704: connect: connection refused` to be a little cryptic.

My suggestion would look like:
```
~$ podman run -it alpine
Error: failed to connect: dial tcp [::1]:49704: connect: connection refused
To start your machine run:

	podman machine start
```

And maybe to run `podman machine init` beforehand, if it's not been done yet.

I just made the switch from `docker` to `podman`, so I thought I'd create a discussion post rather than directly making a PR. I'm not sure how to handle other cases where the same error might happen but for different reasons.

**Comment by Luap99** [comment]
There should be already a better error message in podman:
```
Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM
Error: unable to connect to Podman socket: Get "http://d/v4.6.0/libpod/_ping": dial unix ///run/user/1000/podman/podman.sock: connect: no such file or directory
```

I assume your exact error is not matched for some reason. I would consider this a bug, I can take a look.

**Reply by Luap99** [reply]
should be fixed in https://github.com/containers/podman/pull/18441

**Reply by osalbahr** [reply]
I installed `podman` using `brew`. Could that be why?

**Reply by Luap99** [reply]
Doesn't matter how you install it, the ssh connection error was not matched so podman didn't annotate the error correctly. This will be fixed in the next release once the PR above is merged.

**Reply by osalbahr** [reply]
Thank you for the PR!

I've found the same issue with `podman system info`
```
~$ podman system info
Error: failed to connect: dial tcp [::1]:49704: connect: connection refused
~$ podman --version
podman version 4.5.0
~$ 
```

I don't know how to build `podman` from source to check if the PR fixes this as well.

**Reply by Luap99** [reply]
it will work for all commands
